### ISSUE TITLE
Repeat key not being put/processed correctly in the main buffer

### ISSUE THREAD

**Issue by QRiary** [Ikcelaks/qmk_sequence_transform#36]
If I wanna have `es@t -> essential` rule, since there is no explicit `es@` rule it doesnt work, it wants me to put `esst` there

**Comment by kamih** [comment]
It looks like the `@` key is being removed from the input buffer after handling the default repeat rule:
```
./tester -s "es@t"
--- str[0]: e ---
buffer: |e| (1)
output: |e| (1)
--- str[1]: s ---
buffer: |es| (2)
output: |es| (2)
--- str[2]: @ ---
buffer: |es@| (3)
output: |ess| (3)
--- str[3]: t ---
buffer: |esst| (4)
output: |esst| (4)
```

@Ikcelaks is there a particular reason for this behavior?

**Comment by Ikcelaks** [comment]
This behavior is expected. The idea was to have the repeat key action behave identically to having pressed the repeated key. The idea was to be able to write the rule as `esst` and have it work the same whether you typed it as `esst` or `es@t`

Possibly this is not what we really want.

We can change it so that the `@` stays in the buffer, and is given an action equal to the repeated key

This is probably better now that the fallback buffer is available

**Comment by kamih** [comment]
Yeah, since we now have the fallback buffer, I don't see why we have to make this exception for `@` on default action.

**Comment by Ikcelaks** [comment]
Forgot to say that the best way to fix this currently is to define `es@ -> ess`. Then, we don't over-write the `@` with the `s` in the buffer, and the `es@t -> essential` rule will work.

Changing the current behavior can't be done quite yet, because we currently have no other way to specify in the buffer what output the repeat action is outputing. Later it might be possible

**Comment by Ikcelaks** [comment]
With metacharacters, the repeat action can be defined with the rule `∀👍 ⇒ ∀⑴`. This rule behaves, slightly differently than the existing special output function, but it's arguably what a user would expect. `r*@` used to output `rlr`; with this rule it outputs `rll`.

**Comment by QRiary** [comment]
> With metacharacters, the repeat action can be defined with the rule `∀👍 ⇒ ∀⑴`. This rule behaves, slightly differently than the existing special output function, but it's arguably what a user would expect. `r*@` used to output `rlr`; with this rule it outputs `rll`.

we had [that](<https://discord.com/channels/@me/1207056444134924298/1215876046893875330>) and in #31 discussion already :dofblush:, ig its just the you thing since you are used to built in repeat key implementation, but switch magic behaved exactly like that metachar rule, and its way more intuitive, so I'm totally agreeing there :>
